tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/publicAsset/PublicAssetSetService.scala

```diff
@@ -16,4 +20,15 @@ class PublicAssetSetService @Inject()(val identityRepository: IdentityRepository
         case Seq() => SMono.just(identityIds)
         case noExistIdentityIds => SMono.error(PublicAssetIdentityIdNotFoundException(noExistIdentityIds))
       }
+
+    def create(username: Username, creationRequest: PublicAssetCreationRequest): SMono[PublicAssetStorage] = {
+      SMono(publicAssetRepository.getTotalSize(username))
+        .filter(totalSize => (totalSize + creationRequest.size.value) <= configuration.publicAssetQuotaLimit.asLong())
+        .flatMap(_ => SMono(publicAssetRepository.create(username, creationRequest)))
+        .switchIfEmpty(SMono.error(new PublicAssetTotalSizeExceededException()))
+    }
```

Comment:
The check is still done at the presentation layer.

I would expect it to be done at the storage layer.

Reply:
I don't think `PublicAssetSetService` is a presentation layer.
The `***Repository` should just do simple CRUD logic

Reply:
PublicAssetSetService only wraps part of the storage layer.

If for instance you would ask me to implement a webadmin route for the feature I would clearly not use that.

We tend to put logic in repository. DAO are for simple CRUDs.

Reply:
if we put the check in repository, we will have to duplicate code in every implementations of the repository

Reply:
> if we put the check in repository, we will have to duplicate code in every implementations of the repository

Not true.

We can extract a helper method / class for this.

We have more leverage to make each backend evolve if appropriate.

Also, that's what we do for mailbox quota. I expect the same be done here.

Reply:
moved